### PR TITLE
Add support for using nested DTOs

### PR DESCRIPTION
**Pull request by eltharin**
For https://github.com/doctrine/orm/pull/11540 in good branch

### CODE REVIEW

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @eltharin !

**Review by SenseException — APPROVED** [review]

File: docs/en/reference/dql-doctrine-query-language.rst [hunk 1]
```diff
@@ -579,7 +579,34 @@ And then use the ``NEW`` DQL keyword :
...
+        }
+    }
+	
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW CustomerDTO(c.name, e.email, NEW AddressDTO(a.street, a.city, a.zip)) FROM Customer c JOIN c.email e JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+	
+Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
```
Comment (greg0ire) [inline docs/en/reference/dql-doctrine-query-language.rst:609]: @eltharin sorry, I forgot this during my review, but below there is a paragraph called `EBNF` with a grammar that you should probably update with your changes.

Comment (eltharin) [reply docs/en/reference/dql-doctrine-query-language.rst:609]: OK => https://github.com/doctrine/orm/pull/11619

**Comment by setineos** [comment]
With add support for using nested DTOs, how to use it with `sprintf`?

Currently I have this:

```php
$result = $this->entityManager->createQueryBuilder('c')
    ->select(sprintf(
        'NEW %s(c.name, e.email)',
        CustomerDTO::class
    ))
    ->join('c.email', 'e')
    ->getQuery()
    ->getResult();
```

I would like to add `NEW AddressDTO(a.street, a.city, a.zip)` but with `sprintf` syntax.

**Comment by stetodd** [comment]
Have you tried:

```
    sprintf(
        'NEW %s(c.name, e.email, NEW %s(a.street, a.city, a.zip))',
        CustomerDTO::class,
        AddressDTO::class
    )
```

**Comment by setineos** [comment]
It works perfectly!

**Pull request merged**